### ISSUE TITLE
Test running under SLURM

### ISSUE THREAD

**Issue by peterjc** [pyani-plus/pyani-plus#132]
See https://snakemake.github.io/snakemake-plugin-catalog/plugins/executor/slurm.html which suggests for the command line you just add ``--executor slurm``:

```
$ pip install snakemake-executor-plugin-slurm
$ snakemake --executor slurm ...
```

We will want to run some tests explicitly via slurm (if available), and offer to run via slurm in the public CLI API too.

**Comment by peterjc** [comment]
Also https://github.com/snakemake/snakemake-executor-plugin-slurm/blob/main/docs/further.md looks important, including the bit about using an HPC with environments.

**Comment by peterjc** [comment]
Cross reference snakemake issues:

* https://github.com/snakemake/snakemake/issues/3152 (for local execution, see #138 for our current workaround using ``chronic``)

* https://github.com/snakemake/snakemake/issues/3173 (which makes debugging in quiet mode almost impossible)

**Comment by peterjc** [comment]
Annotated profile for archie-west

```
executor: slurm
jobs: 10
cores: 10
precommand: "module purge && module load miniconda && conda activate pyani-plus_py312 && p
ython --version"

# See https://snakemake.github.io/snakemake-plugin-catalog/plugins/executor/slurm.html#adv
anced-resource-specifications
default-resources:
    # If not specified, get a warning (in purple) then this error (in red):
    # SLURM job submission failed. The error message was
    # sbatch error: Batch job submission failed: No partition specified
    # or system default partition
    slurm_partition: dev

    # If not specified, get the following warning (in purple):
    # No SLURM account given, trying to guess.
    # Guessed SLURM account: pritchard-grxiv
    slurm_account: pritchard-grxiv

    # If not specified, get the following warning (in purple):
    # No wall time information given. This might or might not work
    # on your cluster. If not, specify the resource runtime in your
    # rule or as a reasonable default via --default-resources.
    runtime: 10
```

Will probably want to adjust the runtime (here 10 minutes), and look at ``slurm_init_seconds_before_status_checks`` (default 40s) and similar settings adding to the run time...

**Comment by peterjc** [comment]
Closed via #158 although the wider question of testing and seeing how things scale remains (we expect multiprocessing to win for small runs)
